common/src/main/java/com/scottlogic/deg/common/date/ChronoUnitWorkingDayWrapper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 Scott Logic Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.scottlogic.deg.common.date;
+
+import org.threeten.extra.Temporals;
+
+import java.time.Duration;
+import java.time.Period;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAdjuster;
+import java.time.temporal.TemporalAmount;
+import java.util.function.IntFunction;
+
+public class ChronoUnitWorkingDayWrapper {
+
+    private final ChronoUnit chronoUnit;
+
+    private final boolean workingDay;
+
+    public ChronoUnitWorkingDayWrapper(ChronoUnit chronoUnit, boolean workingDay) {
+        this.chronoUnit = chronoUnit;
+        this.workingDay = workingDay;
+    }
+
+    public TemporalAdjuster adjuster(int value) {
+        return workingDay ? getWorkingDayFunction(value) : getFunctionWithPositivity(chronoUnit, value);
+    }
+
+    private TemporalAdjuster getWorkingDayFunction(int value) {
+        TemporalAdjuster adjuster = value >= 0 ? Temporals.nextWorkingDay() : Temporals.previousWorkingDay();
+        return new RepeatedTemporalAdjuster(adjuster, Math.abs(value));
+    }
+
+    private TemporalAdjuster getFunctionWithPositivity(ChronoUnit unit, int value) {
```

Comment:
Maybe call this `getFunctionWithOffset` because it does positive and negative values?

Reply:
Good call. The intent of this class has shifted quite massively through refactoring.